include/alpaka/core/ApiHipRt.hpp

```diff
@@ -183,10 +183,15 @@ namespace alpaka
             return ::hipFree(devPtr);
         }
 
-        static inline Error_t freeAsync(void* /* devPtr */, Stream_t /* stream */)
+        static inline Error_t freeAsync([[maybe_unused]] void* devPtr, [[maybe_unused]] Stream_t stream)
         {
+            // hipFreeAsync is implemented only in ROCm 5.2.0 and later.
+#    if BOOST_LANG_HIP >= BOOST_VERSION_NUMBER(5, 2, 0)
+            return ::hipFreeAsync(devPtr, stream);
+#    else
```

Comment:
Why not like this?

```suggestion
            // hipFreeAsync is implemented only in ROCm 5.2.0 and later.
    if constexpr(BOOST_LANG_HIP >= BOOST_VERSION_NUMBER(5, 2, 0))
            return ::hipFreeAsync(devPtr, stream);
    else
```

Reply:
both branches of `if constexpr` must be able to _compile_ - which is not the case on older versions of ROCm, where `hipFreeAsync` is not defined.

Reply:
_e.g._ `test.cc`
```c++
#include <iostream>

int main(void) {

  if constexpr(false) {
    does_not_exist();
  } else {
    std::cout << "Hello world\n";
  }

  return 0;
}
```
fails to compile:
```bash
$ g++ test.cc -O2 -std=c++17 -o test
test.cc: In function ‘int main()’:
test.cc:6:5: error: ‘does_not_exist’ was not declared in this scope
    6 |     does_not_exist();
      |     ^~~~~~~~~~~~~~
```

Reply:
Ah, right. My bad.